### PR TITLE
Add BCD for the WebGPU API

### PR DESCRIPTION
**Pull request by chrisdavidmills**
#### Summary

This PR adds BCD for the [WebGPU API](https://gpuweb.github.io/gpuweb). <s>This is expected to land in Chrome 114 or so (around April-ish)</s>.

Update: March 28 2023 — I've confirmed with the Chrome folks that this will be available by default in Chrome 113 beta, which is expected on April 6.

You can see full details of the changes expected in this PR in my [research document](https://docs.google.com/document/d/10wW89xkXN_9fztLUZ9uzKoP9lMbOa02w9dG0qY7geow/edit#heading=h.frl8u3cowbcx). Note that:

### CODE REVIEW

**Comment by chrisdavidmills** [comment]
Note: Confirmed with Google engineering that this API will be available by default in Chrome 113 beta, which is going live later this week.

File: api/GPUExternalTexture.json [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        },
+        "status": {
+          "experimental": true,
+          "standard_track": true,
+          "deprecated": false
+        }
+      },
+      "expired": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/GPUExternalTexture/expired",
```
Comment (Elchi3) [inline api/GPUExternalTexture.json:38]: GPUExternalTexture.expired is non standard and I don't see you creating an MDN page.

File: api/GPUQueue.json [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+            "webview_android": "mirror"
+          },
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": false
+          }
+        }
+      },
+      "onSubmittedWorkDone": {
```
Comment (Elchi3) [inline api/GPUQueue.json:104]: I thought this was an event given the on* name, but looks like it isn't :) Weird naming!

File: api/GPUDevice.json [hunk 3]
```diff
@@ -0,0 +1,857 @@
...
+        }
+      },
+      "features": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/GPUDevice/features",
+          "spec_url": "https://gpuweb.github.io/gpuweb/#dom-gpudevice-features",
+          "support": {
+            "chrome": {
+              "version_added": "113",
+              "notes": "Not all features will be available, even if supported by the underlying hardware, due to constraints in the browser or adapter."
```
Comment (Elchi3) [inline api/GPUDevice.json:553]: See previous comment

File: api/GPUDevice.json [hunk 4]
```diff
@@ -0,0 +1,857 @@
...
+        }
+      },
+      "limits": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/GPUDevice/limits",
+          "spec_url": "https://gpuweb.github.io/gpuweb/#dom-gpudevice-limits",
+          "support": {
+            "chrome": {
+              "version_added": "113",
+              "notes": "Some limits will be fixed at a base or tiered value, regardless of the value requested."
```
Comment (Elchi3) [inline api/GPUDevice.json:656]: See previous comment

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Heroic work, Mr Mills! 👍 

I believe the intent to ship is for Chrome Desktop only. This means all Chromium-based browsers that are not Desktop browsers should be set to `false` instead of `"mirror"`. 

Also, given how gigantic this API is, are we sure other desktop browsers like Edge will just enable it, too?

Additional inline comments. Thanks!

File: api/GPUAdapter.json [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+        }
+      },
+      "limits": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/GPUAdapter/limits",
+          "spec_url": "https://gpuweb.github.io/gpuweb/#dom-gpuadapter-limits",
+          "support": {
+            "chrome": {
+              "version_added": "113",
+              "notes": "Chrome reports different tier values of different limits rather than the exact limits of each GPU, to reduce the unique information available to drive-by fingerprinting."
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply api/GPUAdapter.json:112]: removed

File: api/GPUAdapter.json [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+        }
+      },
+      "requestDevice": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/GPUAdapter/requestDevice",
+          "spec_url": "https://gpuweb.github.io/gpuweb/#dom-gpuadapter-requestdevice",
+          "support": {
+            "chrome": {
+              "version_added": "113",
+              "notes": [
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply api/GPUAdapter.json:181]: removed

File: api/GPUExternalTexture.json (hunk 1 above)
Comment (chrisdavidmills) [reply api/GPUExternalTexture.json:38]: Ah, good catch; this is one of the non-standard things I didn't end up removing. removed now.

File: api/GPUQueue.json (hunk 2 above)
Comment (chrisdavidmills) [reply api/GPUQueue.json:104]: Yup, I thought that too

File: api/GPUDevice.json (hunk 3 above)
Comment (chrisdavidmills) [reply api/GPUDevice.json:553]: removed

File: api/GPUDevice.json (hunk 4 above)
Comment (chrisdavidmills) [reply api/GPUDevice.json:656]: removed

**Comment by chrisdavidmills** [comment]
Thanks for the review @Elchi3 ! I've made fixes for your individual comments. As for the wider support questions, I've asked a few people who will probably know, and will make updates as I get answers.

For Chromium-based browsers where we are just not sure yet, I'm wondering whether it'd be better to just mark them as "no" for now, and then we can update the info as we get more data.

**Comment by chrisdavidmills** [comment]
@Elchi3 OK, mammoth data update done ;-)

**Review by Elchi3 — APPROVED** [review]
Amazing! Well done, GPU-Chris! 🎉

**Comment by chrisdavidmills** [comment]
> Amazing! Well done, GPU-Chris! 🎉

LOL, I now know way more about that API than I ever thought I would! Thanks for handling the review mate.

**Pull request merged**